finalitygadget/finalitygadget.go

```diff
@@ -499,43 +574,116 @@ func (fg *FinalityGadget) queryBlockByHeight(blockNumber int64) (*types.Block, e
 	}, nil
 }
 
-func (fg *FinalityGadget) handleBlock(ctx context.Context, latestFinalizedHeight uint64) error {
-	for height := fg.lastProcessedHeight + 1; height <= latestFinalizedHeight; height++ {
+type blockResult struct {
+	height      uint64
+	block       *types.Block
+	isFinalized bool
+	err         error
+}
+
+func (fg *FinalityGadget) processBlocksTillHeight(ctx context.Context, latestHeight uint64) error {
+	for startHeight := fg.lastProcessedHeight + 1; startHeight <= latestHeight; {
```

Comment:
`startHeight` sounds like at the level of the whole function. i was gonna suggest to rename to `currentHeight` but then I realize it's for a batch.

so `batchStartHeight` and `batchEndHeight` seems to be better names